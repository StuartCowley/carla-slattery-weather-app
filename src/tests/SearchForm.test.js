import React from "react";
import { fireEvent, screen, render } from "@testing-library/react";

import SearchForm from "../components/SearchForm";

describe("SearchForm", () => {
  const validProps = {
    searchText: "Liverpool",
    setSearchText: jest.fn(),
    onSubmit: jest.fn(),
  };

  it("renders correctly", () => {
    const { asFragment } = render(
      <SearchForm
        searchText={validProps.searchText}
        setSearchText={validProps.setSearchText}
        onSubmit={validProps.onSubmit}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it("renders an input text search box", () => {
    const { getByRole } = render(<SearchForm {...validProps} />);
    expect(getByRole("textbox")).toBeInTheDocument();
  });
  it("calls event handler function after clicking submit", () => {
    render(<SearchForm {...validProps} />);
    fireEvent.click(screen.getByRole("button"));
    expect(validProps.onSubmit).toHaveBeenCalled();
  });
});
