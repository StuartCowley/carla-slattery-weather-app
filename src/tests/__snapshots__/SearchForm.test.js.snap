// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm renders correctly 1`] = `
<DocumentFragment>
  <div
    class="search-form"
  >
    <input
      placeholder="Enter Location"
      type="text"
      value="Liverpool"
    />
    <button
      type="submit"
    >
      Search
    </button>
  </div>
</DocumentFragment>
`;
